### PR TITLE
[#1880] Change hasAuthorConfigFile to become a local variable

### LINKED ISSUE #1880: Changing `AuthorConfiguration.hasAuthorConfigFile` into a local variable
Possibly related to #1879, `AuthorConfiguration.hasAuthorConfigFile` is currently a global variable that determines whether during the analysis of repo files, a new author should be added and tracked. If an author config file is provided, an unrecognized author is ignored, otherwise it is added and tracked. 

This might cause parallel runs of test cases at the method level, especially system test cases, to have issues. A suggestion would be that in the `RepoConfiguration::merge` method, we can individually set the `AuthorConfig` inside each `RepoConfig` file to have the local variable flag `hasAuthorConfigFile` as true since the method is only ever called if an author config csv is specified.

### PR DESCRIPTION
**Pull request by sikai00**
Fixes #1880 
Part of #1879
Part of #1873 

### Proposed commit message

```
AuthorConfiguration.hasAuthorConfigFile is currently a static variable 
that determines whether during the analysis of repo files, a new author
should be added and tracked. If an author config file is provided, an
unrecognized author is ignored, otherwise it is added and tracked.

This might cause parallel runs of test cases at the method level, 
especially system test cases, to have issues.

Let's make AuthorConfiguration.hasAuthorConfigFile a local variable to
the AuthorConfiguration found within each RepoConfiguration.
```

### Other information

As mentioned by @chan-j-d, this issue is indeed related to #1879. This PR has to be merged in before the PR relevant to #1879. This is the case because in that PR, `AuthorConfiguration.hasAuthorConfigFile` is no longer set to false for all systemtest. LocalConfigSystemTest will fail as a result, as it does not have author-config.csv, however due to the global variable RepoSense believes it does.

This is also highly relevant to parallelizing of the ConfigSystemTest, as these static variables are shared across the threads used to run each test in ConfigSystemTest.

### CODE REVIEW

**Review by ckcherry23 — APPROVED** [review]
LGTM!

File: src/main/java/reposense/model/AuthorConfiguration.java [hunk 1]
```diff
@@ -329,11 +329,11 @@ public boolean containsName(String name) {
...
-    public static void setHasAuthorConfigFile(boolean hasAuthorConfigFile) {
-        AuthorConfiguration.hasAuthorConfigFile = hasAuthorConfigFile;
+    public void setHasAuthorConfigFile(boolean hasAuthorConfigFile) {
+        this.hasAuthorConfigFile = hasAuthorConfigFile;
     }
 
-    public static boolean hasAuthorConfigFile() {
-        return hasAuthorConfigFile;
+    public boolean hasAuthorConfigFile() {
+        return this.hasAuthorConfigFile;
```
Comment (chan-j-d) [inline src/main/java/reposense/model/AuthorConfiguration.java:337]: Minor point but iirc we don't use 'this' if the reference is not ambiguous, so just `return hasAuthorConfigFile`.

Comment (sikai00) [reply src/main/java/reposense/model/AuthorConfiguration.java]: Okay, I've updated it, thanks for noticing that!

**Review by chan-j-d — APPROVED** [review]
LGTM!

**Review by yhtMinceraft1010X — APPROVED** [review]
LGTM

**Pull request merged**